src/user.rs

```diff
@@ -297,28 +301,110 @@ impl User<Synching> {
         };
 
         let room_id = &rooms[room];
-        let content = AnyMessageEventContent::RoomMessage(MessageEventContent::text_plain(
+        let content = AnyMessageLikeEventContent::RoomMessage(RoomMessageEventContent::text_plain(
             get_random_string(),
         ));
         let instant = Instant::now();
-        let response = client.room_send(room_id, content, None).await;
-
-        match response {
-            Ok(response) => {
-                self.send(Event::RequestDuration((
-                    UserRequest::SendMessage,
-                    instant.elapsed(),
-                )))
-                .await;
-
-                self.send(Event::MessageSent(response.event_id.to_string()))
+        if let Some(room) = client.get_joined_room(room_id) {
+            let response = room.send(content, None).await;
+            match response {
+                Ok(response) => {
+                    self.send(Event::RequestDuration((
+                        UserRequest::SendMessage,
+                        instant.elapsed(),
+                    )))
                     .await;
-            }
-            Err(e) => {
-                if let matrix_sdk::Error::Http(e) = e {
-                    self.send(Event::Error((UserRequest::SendMessage, e))).await;
+
+                    self.send(Event::MessageSent(response.event_id.to_string()))
+                        .await;
+                }
+                Err(e) => {
+                    if let matrix_sdk::Error::Http(e) = e {
+                        self.send(Event::Error((UserRequest::SendMessage, e))).await;
+                    }
                 }
             }
         }
     }
 }
+
+async fn on_room_message(
+    event: OriginalSyncRoomMessageEvent,
+    room: Room,
+    sender: Sender<Event>,
+    user_id: Box<UserId>,
+) {
+    if let Room::Joined(room) = room {
+        if event.sender.localpart() == user_id.localpart() {
+            return;
+        }
+        sender
+            .send(Event::MessageReceived(event.event_id.to_string()))
+            .await
+            .expect("Receiver dropped");
+        log::info!(
+            "User {} received a message from room {} and sent by {}",
+            user_id,
+            room.room_id(),
+            event.sender
+        );
+    }
+}
+
+// /// This function returns homeserver domain and url, ex:
+// ///  - get_homeserver_url("matrix.domain.com") => ("matrix.domain.com", "https://matrix.domain.com")
```

Comment:
Let's leave the comments as actual documentation for the function.
```suggestion
/// This function returns homeserver domain and url, ex:
///  - get_homeserver_url("matrix.domain.com") => ("matrix.domain.com", "https://matrix.domain.com")
```